includes/Core/Modules/Modules.php

```diff
@@ -428,6 +430,31 @@ function( Module $module ) {
 		return $data;
 	}
 
+	/**
+	 * Populates modules data to pass to JS.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $modules_data Inline modules data.
+	 * @return array Inline modules data.
+	 */
+	private function inline_modules_data( $modules_data ) {
+		$available_modules = $this->get_available_modules();
+
+		$modules_with_data_available_state = array_filter(
+			$available_modules,
+			function( Module $module ) {
+				return $module instanceof Module_With_Data_Available_State;
+			}
+		);
+
+		foreach ( $modules_with_data_available_state as $module ) {
```

Comment:
Nothing wrong with this way but it could be simplified into a single `foreach` over available modules and only set the data available state if the module implements the interface.